src/config.rs

```diff
@@ -0,0 +1,104 @@
+use std::{collections::HashMap, ffi::OsStr, fs};
+
+use common::extensions::*;
+use microtemplate::render;
+use serde::{Deserialize, Deserializer};
+use uniquote::Quote;
+
+use crate::paths::PATHS;
+
+#[derive(Clone, Default, Debug)]
+pub struct ConfigContext;
+
+impl microtemplate::Context for ConfigContext {
+    fn get_field(&self, field_name: &str) -> &str {
+        match field_name {
+            "game_dir" => PATHS.game_str.as_str(),
+            _ => "",
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ModConfig {
+    #[serde(default)]
+    pub(crate) args: HashMap<String, String>,
+    #[serde(default)]
+    pub(crate) tasks: Vec<Task>,
+}
+
+fn default_as_false() -> bool {
+    false
+}
+
+fn default_as_true() -> bool {
+    true
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Task {
+    pub(crate) command: String,
+    #[serde(default = "default_as_false")]
+    pub(crate) terminate_on_errors: bool,
+    #[serde(default = "default_as_true")]
+    pub(crate) no_window: bool,
+    #[serde(default)]
+    pub(crate) args: Vec<String>,
+    #[serde(flatten, deserialize_with = "deserialize_render_map")]
+    pub(crate) values: HashMap<String, String>,
+}
+
+fn deserialize_render_map<'de, D>(deserializer: D) -> Result<HashMap<String, String>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let hash_map = HashMap::<String, String>::deserialize(deserializer)?;
+    Ok(hash_map
+        .into_iter()
+        .map(|(key, value)| (key, render(value.as_str(), ConfigContext {})))
+        .collect())
+}
```

Comment:
decoding shouldn't be coupled with templating, it'd be cleaner to just to standard deserialization and render the template later

Reply:
This is related to the odd thing I did to ConfigContext. See my [comment](https://github.com/jac3km4/cybercmd/pull/2#discussion_r1142600016) there 🙂